fabric-tool-attribute-api-v1/src/main/java/net/fabricmc/fabric/mixin/tool/attribute/MixinItemStack.java

```diff
@@ -31,40 +31,26 @@
 import net.minecraft.item.ItemStack;
 
 import net.fabricmc.fabric.api.tool.attribute.v1.DynamicAttributeTool;
-import net.fabricmc.fabric.api.util.TriState;
+import net.fabricmc.fabric.api.tool.attribute.v1.ToolManager;
 import net.fabricmc.fabric.impl.tool.attribute.AttributeManager;
-import net.fabricmc.fabric.impl.tool.attribute.ToolManager;
 
 @Mixin(ItemStack.class)
 public abstract class MixinItemStack {
 	@Shadow
 	public abstract Item getItem();
 
-	@Inject(at = @At("HEAD"), method = "isEffectiveOn", cancellable = true)
+	@Inject(at = @At("RETURN"), method = "isEffectiveOn", cancellable = true)
 	public void isEffectiveOn(BlockState state, CallbackInfoReturnable<Boolean> info) {
-		TriState triState = ToolManager.handleIsEffectiveOn((ItemStack) (Object) this, state, null);
-
-		if (triState != TriState.DEFAULT) {
-			info.setReturnValue(triState.get());
-			info.cancel();
+		if (!info.getReturnValueZ()) {
+			info.setReturnValue(ToolManager.handleIsEffectiveOnIgnoresVanilla(state, (ItemStack) (Object) this, null).get());
```

Comment:
Is it ok to not have this be TriStates anymore?

Reply:
Yeah I should do that